nexus/db-queries/src/db/datastore/volume.rs

```diff
@@ -203,6 +218,244 @@ impl DataStore {
             .map_err(|e| public_error_from_diesel(e, ErrorHandler::Server))
     }
 
+    async fn volume_checkout_allowed(
```

Comment:
This looks great. Thanks for reorganizing this--I think it's much easier to read in this form.